content/en/docs/concepts/workloads/controllers/job.md

```diff
@@ -568,6 +568,56 @@ The new Job itself will have a different uid from `a8f3d00d-c6d2-11e5-9f87-42010
 `manualSelector: true` tells the system to that you know what you are doing and to allow this
 mismatch.
 
+### Job tracking with finalizers
+
+{{< feature-state for_k8s_version="v1.22" state="alpha" >}}
+
+{{< note >}}
+In order to use this behavior, you must enable the `JobTrackingWithFinalizers`
+[feature gate](/docs/reference/command-line-tools-reference/feature-gates/)
+on the [API server](/docs/reference/command-line-tools-reference/kube-apiserver/)
+and the [controller manager](/docs/reference/command-line-tools-reference/kube-controller-manager/).
+It is disabled by default.
+
+When enabled, the control plane tracks new Jobs using the behavior described
+below. Existing Jobs are unaffected. Users don't interact with this feature.
+{{< /note >}}
+
+When this feature isn't enabled, the Job {{< glossary_tooltip term_id="controller" >}}
+relies on counting the Pods that exist in the cluster to track the Job status,
+that is, to keep the counters for `succeeded` and `failed` Pods.
+However, Pods can be removed for a number of reasons, including:
+- The garbage collector that removes orphan Pods when a Node goes down.
+- The garbage collector that removes finished Pods (in `Succeeded` or `Failed`
+  phase) after a threshold.
+- Human intervention to delete Pods belonging to a Job.
+- An external controller (not provided as part of Kubernetes) that removes or
+  replaces Pods.
+
+If you enable the `JobTrackingWithFinalizers` feature for your cluster, the
+control plane keeps track of the Pods that belong to any Job and notices if any
+such Pod is removed from the API server. To do that, the Job controller creates Pods with
+the finalizer `batch.kubernetes.io/job-tracking`. The controller removes the
+finalizer only after the Pod has been accounted for in the Job status, allowing
+the Pod to be removed by other controllers or users.
+
+The Job controller uses the new algorithm for new Jobs only. Jobs created
+before the feature is enabled are unaffected. You can determine if the Job
+controller is tracking a Job using finalizers if they have the annotation
+`batch.kubernetes.io/job-tracking`. You should **not** manually add or remove
+this annotation from Jobs.
+
+#### Transition to stable feature {#job-tracking-with-finalizers-graduation}
+
+The `JobTrackingWithFinalizers` feature will graduate to
+[general availability](/docs/reference/command-line-tools-reference/feature-gates/#feature-stages) (GA),
+or might potentially be deprecated if testing reveals a design defect.
+It's likely that this feature would reach general availability, and in that case then the
+`batch.kubernetes.io/job-tracking` annotation will from that point remain honored
+for at least 2 further minor Kubernetes releases. After _at least_
+those 2 further releases, the `batch.kubernetes.io/job-tracking` annotation would become
+deprecated and Kubernetes will switch to tracking completion for all Jobs using finalizers.
+
```

Comment:
This subsection is about future, so it can be safely removed.

Reply:
Sg. Removed